File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # for i in range(len(numbers)-1):
        #     for j in range(i+1,len(numbers)):
        #         if numbers[i] + numbers[j] == target:
        #             return[i+1, j+1]
        # return []        

        left, right = 0, len(numbers)-1
        while left < right:
            res = numbers[left] + numbers[right]
            if res == target:
                break
            elif res < target:
                left +=1
            else:
                right -=1

        return [left + 1, right + 1]

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private int res = 0;
    private void dfs(TreeNode node, int depth){
        if(node == null) return;
        depth++;
        res = Math.max(depth, res);
        dfs(node.left,depth);
        dfs(node.right, depth);
    }
    public int maxDepth(TreeNode root) {
        dfs(root, 0);
        return res;        

    }
}
